index.js

```diff
@@ -10,25 +10,23 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
-		return thunk;
 
-		function thunk(result) {
-			if (promise) {
-				// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
-				promise.catch(function () {});
-			}
+		// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
+		promise.catch(function () {});
```

Comment:
In short, either we catch all of them, or catch none. This PR catches all of them, the stable code catches them only if the rejection happens not much after the `doSomethingFor5000ms()` ends, i.e.:

```js
// caught as expected
delay(5)
  .then(delay.reject(10))
  .catch(function () {
    console.log('all good!')
  });

// `unhandledrejection` and then `rejectionhandled`

delay(20)
  .then(delay.reject(10))
  .catch(function () {
    console.log('all good!')
  });
```